services/QuillLMS/app/controllers/concerns/growth_results_summary.rb

```diff
@@ -0,0 +1,151 @@
+        description: skill_group.description,
+        not_yet_proficient_in_pre_test_student_names: [],
+        not_yet_proficient_in_post_test_student_names: []
+      }
+    end
+
+    {
+      student_results: student_results,
+      skill_group_summaries: @skill_group_summaries
+    }
+  end
+
+  private def set_pre_test_activity_sessions_and_assigned_students(activity_id, classroom_id)
+    unit_ids = @current_user.units.joins("JOIN unit_activities ON unit_activities.activity_id = #{activity_id}")
+    classroom_units = ClassroomUnit.where(unit_id: unit_ids, classroom_id: classroom_id)
+    assigned_student_ids = classroom_units.map { |cu| cu.assigned_student_ids }.flatten.uniq
+    @pre_test_assigned_students = User.where(id: assigned_student_ids).sort_by { |u| u.last_name }
+    @pre_test_activity_sessions = ActivitySession.where(activity_id: activity_id, classroom_unit_id: classroom_units.ids, state: 'finished').order(completed_at: :desc).uniq { |activity_session| activity_session.user_id }
+  end
+
+  private def set_post_test_activity_sessions_and_assigned_students(activity_id, classroom_id)
+    unit_ids = @current_user.units.joins("JOIN unit_activities ON unit_activities.activity_id = #{activity_id}")
+    classroom_units = ClassroomUnit.where(unit_id: unit_ids, classroom_id: classroom_id)
+    assigned_student_ids = classroom_units.map { |cu| cu.assigned_student_ids }.flatten.uniq
+    @post_test_assigned_students = User.where(id: assigned_student_ids).sort_by { |u| u.last_name }
+    @post_test_activity_sessions = ActivitySession.where(activity_id: activity_id, classroom_unit_id: classroom_units.ids, state: 'finished').order(completed_at: :desc).uniq { |activity_session| activity_session.user_id }
+  end
+
+  private def student_results
+    @post_test_assigned_students.map do |assigned_student|
+      post_test_activity_session = @post_test_activity_sessions.find { |as| as.user_id == assigned_student.id }
+      pre_test_activity_session = @pre_test_activity_sessions.find { |as| as.user_id == assigned_student.id }
+      if post_test_activity_session
+        skill_groups = skill_groups_for_session(@skill_groups, post_test_activity_session.id, pre_test_activity_session.id, assigned_student.name)
+        total_acquired_skills_count = skill_groups.map { |sg| sg[:acquired_skill_ids] }.flatten.uniq.count
+        {
+          name: assigned_student.name,
+          id: assigned_student.id,
+          skill_groups: skill_groups,
+          total_acquired_skills_count: total_acquired_skills_count
+        }
+      else
+        { name: assigned_student.name }
+      end
+    end
+  end
+
+  private def data_for_skill_by_activity_session(activity_session_id, skill)
+    concept_results = ConceptResult.where(activity_session_id: activity_session_id, concept_id: [skill.concept_ids])
+    number_correct = concept_results.select(&:correct?).length
+    number_incorrect = concept_results.reject { |cr| cr.correct? }.length
```

Comment:
Should be able to use `reject(&:correct?)` here the same way you used it for `select` immediately above.